djpsa/halo/records/budgetdata/sync.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from djpsa.halo import models
+from djpsa.halo.records import api
+from djpsa.halo import sync
+
+
+logger = logging.getLogger(__name__)
+
+
+class BudgetDataSynchronizer(sync.HaloSynchronizer):
+    model_class = models.BudgetDataTracker
+    client_class = api.TicketAPI
+
+    related_meta = {
+        'ticket_id': (models.Ticket, 'ticket'),
+        'budgettype_id': (models.BudgetType, 'budget_type'),
+    }
+
+    def fetch_records(self, results, params=None):
+        batch = 1
+        for ticket in models.Ticket.objects.all().values_list(
+                'id', flat=True):
+            logger.info(
+                'Fetching {} records, batch {}'.format(
+                    self.get_model_name(), batch)
+            )
+            response = self.client.fetch_resource(
```

Comment:
I think the client ought to worry about what the URL is. The synchronizer can just worry about what records to sync and when.